src/main/diagnostics/diagnostics.ts

```diff
@@ -0,0 +1,48 @@
+import yaml from 'yaml';
+
+import DIAGNOSTICS_TABLE from '@/assets/diagnostics.yaml';
+
+type DiagnosticsFix = {
+  fixes: Record<'description', string>
+};
+
+type DiagnosticsCheck = {
+  id: string,
+  documentation: string,
+  description: string,
+  mute: boolean,
+  fixes: DiagnosticsFix,
+};
+
+type DiagnosticsCategories = {
+  title: string,
+  checks: Array<DiagnosticsCheck>,
+};
+
+type DiagnosticsType = {
+  last_update: string
+  categories: Array<DiagnosticsCategories>,
+};
+
+export class Diagnostics {
+  diagnostics: DiagnosticsType;
+  checksByCategory: Record<string, Array<DiagnosticsCheck>> = {};
+  constructor(diagnosticsTable: DiagnosticsType|undefined = undefined) {
+    this.diagnostics = diagnosticsTable || DIAGNOSTICS_TABLE.diagnostics;
+    for (const category of this.diagnostics.categories) {
+      this.checksByCategory[category.title] = category.checks;
+    }
+  }
+
+  getCategoryNames() {
+    return Object.keys(this.checksByCategory);
```

Comment:
This means that the category names are not actually sorted, right? Should that be documented? (A JSDoc for all the public methods would be nice for details like this.)

Reply:
jsdoc added. I'm not sure whether this function should sort the names, or leave that to the caller. So putting that off for now (remember there's more work to do, this item is to get the stakes in the ground, although we do need to get the right stakes in the right spots).

Reply:
Note that the original YAML has a defined order (it's a sequence, not a mapping); however, when we read it we discard the ordering.

At a minimum, we should at least document that the order is undefined.